src/features/agent-dashboard/pages/UpdateAgentView.tsx

```diff
@@ -10,24 +10,38 @@ import { PageCrumb, PageHeader, SmallContainer } from 'common/styles/page';
 import { FC, useEffect, useState } from 'react';
 import { Link, useNavigate, useParams } from 'react-router-dom';
 import { AgentDetailForm, FormData } from '../components/AgentDetailForm';
+import { useAuth } from 'features/auth/hooks';
+import { ChangeLog } from 'common/components/ChangeLog';
 
 export type RouteParams = {
   id: string;
 };
 
 export const UpdateAgentView: FC = () => {
   const { id } = useParams<RouteParams>();
+  const { user } = useAuth();
   const navigate = useNavigate();
   const [updateAgent] = useUpdateAgentMutation();
   const { data: agent, isLoading: isLoadingAgent, isFetching, error } = useGetAgentByIdQuery(id!);
+  const { data: agentHistory, error: agentHistoryError } = useGetAgentHistoryQuery(
+    { id, page: 1, pageSize: 5 },
+    { skip: user?.role !== 'ADMIN' },
+  );
   const [formValidationErrors, setFormValidationErrors] = useState<ServerValidationErrors<FormData> | null>(null);
 
   useEffect(() => {
     if (error) {
       notificationService.showErrorMessage('Unable to load agent. Returning to agent list.');
       navigate('/agents', { replace: true });
     }
-  }, [error, navigate]);
+    if (agentHistoryError) {
+      notificationService.showErrorMessage("Unable to load the agent's change history.");
+    }
+  }, [error, navigate, agentHistoryError]);
+
+  const handleShowAllChanges = () => {
+    // console.log('handleShowAllChanges');
```

Comment:
Lets display a modal if they want to see all changes, that displays either a paginated or infinite scroll list of changes.